metaclass&descriptor/tests.py

```diff
@@ -0,0 +1,268 @@
+import ast
+import copy
+import json
+import pytest
+from shop_emulator import ShopWarehouse, PriceDescriptor, DatabaseOpenCloseMixin, CustomerBasket
+import os
+import shutil
+import simplejson
+from config_test import *
+
+
+@pytest.fixture(scope="module")
+def copy_base_file():
+    """ Fixture backuped main database files and replaced to mock database """
+    shutil.copyfile(os.path.abspath("shopDataBase.json"),
+                    os.path.join(os.getcwd(), "temporary_storage_master/shopDataBase.json"))
+    shutil.copyfile(os.path.abspath("shopDataBaseBackup.json"),
+                    os.path.join(os.getcwd(), "temporary_storage_master/shopDataBaseBackup.json"))
+    shutil.copyfile(os.path.join(os.getcwd(), "files_for_test_open_close/shopDataBaseMock.json"),
+                    os.path.abspath("shopDataBase.json"))
+    yield
+    shutil.copyfile(os.path.join(os.getcwd(), "temporary_storage_master/shopDataBase.json"),
+                    os.path.abspath("shopDataBase.json"))
+    shutil.copyfile(os.path.join(os.getcwd(), "temporary_storage_master/shopDataBaseBackup.json"),
+                    os.path.abspath("shopDataBaseBackup.json"))
+
+
+class TestsDescriptor():
+    """Test class tested description PriceDescriptor """
+    class MockClass(): # created mock class with PriceDescriptor
+        price = PriceDescriptor()
+
+        def __init__(self, price):
+            self.price = price
+
+    @pytest.mark.parametrize("price,price_nds", [(15.0, 18.000), (25.0, 30.000)])
+    def test_set(self, price, price_nds):
+        """ test set new value in price attribute"""
+        desc = PriceDescriptor()
+        res = desc.__set__(desc, price)
+        assert res == price_nds
+
+    @pytest.mark.parametrize("price,price_nds", [(15.0, 18.000), (25.0, 30.000)])
+    def test_get(self, price, price_nds):
+        """test for getting value attribute price"""
+        unit = self.MockClass(price)
```

Comment:
you should test your class, not MockClass with the same functionality as your class have